manager/integration/tests/test_basic.py

```diff
@@ -774,8 +776,178 @@ def test_backup_metadata_deletion():  # NOQA
     20. verify that volume(1) has been deleted in the backupstore.
     21. cleanup
     """
-    pass
+    backupstore_cleanup(client)
+
+    volume1_name = volume_name + "-1"
+    volume2_name = volume_name + "-2"
+
+    host_id = get_self_host_id()
+
+    volume1 = create_and_check_volume(client, volume1_name)
+    volume2 = create_and_check_volume(client, volume2_name)
+
+    volume1.attach(hostId=host_id)
+    volume2.attach(hostId=host_id)
+
+    volume1 = wait_for_volume_healthy(client, volume1_name)
+    volume2 = wait_for_volume_healthy(client, volume2_name)
+
+    v1bv, v1b1, _, _ = create_backup(client, volume1_name)
+    v2bv, v2b1, _, _ = create_backup(client, volume2_name)
+    _, v1b2, _, _ = create_backup(client, volume1_name)
+    _, v2b2, _, _ = create_backup(client, volume2_name)
+
+    bvs = client.list_backupVolume()
+
+    for bv in bvs:
+        backups = bv.backupList()
+        for b in backups:
+            assert b.messages is None
+
+    v1b1_new = v1bv.backupGet(name=v1b1.name)
```

Comment:
Do we have to do this check on the returned backups from the backupList?
Also consider adding a helper function like assert_backup_state(actualState, expectedState)
where actualState maps to v1b1_new and expectedState maps to v1b1 in the below code.
That will make the test function more readable :)

Reply:
fixed